src/utils/callGraph.ts

```diff
@@ -0,0 +1,163 @@
+import _ from 'lodash';
+import { HermesSample } from '../types/hermesProfileInterfaces';
+import { SharedEventProperties } from '../types/chromeEventInterfaces';
+import { EventsPhase } from '../types/Phases';
+export class Node {
+  children: Node[];
+  constructor(public id: number, public events: HermesSample[]) {
+    this.children = [];
+  }
+}
+
+export class CallGraph {
+  constructor(public root: Node) {}
+  /* Given an Id find the node corresponding to it */
+  findNode(id: string, node: Node): Node | undefined {
+    if (node.id === parseInt(id)) {
+      return node;
+    }
+    for (let i = 0; i < node.children.length; i++) {
+      const child: Node = node.children[i];
+      const res = this.findNode(id, child);
+      if (res) {
+        return res;
+      }
+    }
+    return undefined;
+  }
+  /*
+    We can only add a node to the call graph if its parent exists.
+    Nodes correspond to stackFrames
+  */
+  addNode(nodeId: string, parentId: string): boolean {
```

Comment:
Nodes can be added to our graph, if we know the id we want to add and its parent. 

Traverses through the graph and find the parentNode. Upon finding the parent Node we simply append the new node, to the children of this parent node

One concern here is that the parent may not exist before we try to add the child. This however has not come up in our problem (verified using the print nodes method later defined).